src/datatype.c

```diff
@@ -930,8 +930,21 @@ static void init_struct_tail(jl_datatype_t *type, jl_value_t *jv, size_t na)
 JL_DLLEXPORT jl_value_t *jl_new_structv(jl_datatype_t *type, jl_value_t **args, uint32_t na)
 {
     jl_ptls_t ptls = jl_get_ptls_states();
-    if (!jl_is_datatype(type) || type->layout == NULL)
+    if (!jl_is_datatype(type) || type->layout == NULL) {
+        // As a special case we're allowed to have unionalls over yakc,
```

Comment:
Can we deal with this elsewhere? It would be better if `new` only has to worry about allocating the type passed to it, instead of also determining what type to allocate.

Reply:
Yes, sorry. This is outdated and can be removed.